File: go/update_credentials.go
package main

import (
	"fmt"
	"os"
)

func main() {
	trunk := os.Getenv("TRUNK_SID")
	cid := os.Getenv("CREDENTIAL_ID")
	if trunk == "" || cid == "" {
		fmt.Println("Error: TRUNK_SID and CREDENTIAL_ID required")
		os.Exit(1)
	}
	body := map[string]string{}
	if v := os.Getenv("SIP_CRED_FRIENDLY_NAME"); v != "" {
		body["friendly_name"] = v
	}
	if v := os.Getenv("SIP_CRED_USERNAME"); v != "" {
		body["user_name"] = v
	}
	if v := os.Getenv("SIP_CRED_PASSWORD"); v != "" {
		body["password"] = v
	}
	if len(body) == 0 {
		body["friendly_name"] = "updated_label"
	}
	fmt.Println("Updating credential...")
	put("/trunks/"+trunk+"/credentials/"+cid, body)
}

File: php/update_credentials.php
#!/usr/bin/env php
<?php
require '_client.php';
$trunk = getenv('TRUNK_SID');
$cid = getenv('CREDENTIAL_ID');
if (!$trunk || !$cid) {
    echo "Error: TRUNK_SID and CREDENTIAL_ID required\n";
    exit(1);
}
$body = [];
if (getenv('SIP_CRED_FRIENDLY_NAME')) {
    $body['friendly_name'] = getenv('SIP_CRED_FRIENDLY_NAME');
}
if (getenv('SIP_CRED_USERNAME')) {
    $body['user_name'] = getenv('SIP_CRED_USERNAME');
}
if (getenv('SIP_CRED_PASSWORD')) {
    $body['password'] = getenv('SIP_CRED_PASSWORD');
}
if (!$body) {
    $body['friendly_name'] = 'updated_label';
}
echo "Updating credential...\n";
exo_put("/trunks/$trunk/credentials/$cid", $body);
